R/utilities-mpi-error-handling.R

```diff
@@ -0,0 +1,153 @@
+    success <- TRUE
+  }
+  return(success)
+}
+
+
+#' @title checkNotNull
+#' @description Check that an object is not null.
+#' @param object to be checked if not null
+#' @export
+checkNotNull <- function(object) {
+  success <- FALSE
+  if (!is.null(object)) {
+    success <- TRUE
+  }
+  return(success)
+}
+
+#' @title checkAllCoresSuccessful
+#' @description Check that all cores executed an mpi.remote.exec command successfully.
+#' @param coreResults list of results returned by each core after an mpi.remote.exec command is complete
+checkAllCoresSuccessful <- function(coreResults) {
+  allCoreResults <- sapply(1:length(coreResults), function(n, obj) obj[[n]], coreResults)
+  return(all(allCoreResults))
+}
+
+#' @title loadSimulationOnCores
+#' @description Send structureSet to core, check its simulation has been loaded successfully
+#' @param structureSet containing simulationSet which contains path to simulation file and pathIDs to be loaded in simulation object as outputs
+loadSimulationOnCores <- function(structureSet, logFolder = getwd()) {
+  success <- FALSE
+  Rmpi::mpi.bcast.Robj2slave(obj = structureSet)
+  Rmpi::mpi.remote.exec(sim <- loadSimulationWithUpdatedPaths(structureSet$simulationSet))
+  simulationLoaded <- Rmpi::mpi.remote.exec(checkSimulationLoaded(simulation = sim, simulationFilePath = structureSet$simulationSet$simulationFile))
+  success <- checkAllCoresSuccessful(simulationLoaded)
+  validateIsLogical(success)
+  if (success) {
+    logWorkflow(message = paste("Simulation file", structureSet$simulationSet$simulationFile, "loaded successfully on all cores"), pathFolder = logFolder)
+  } else {
+    logErrorThenStop(message = paste("Simulation file", structureSet$simulationSet$simulationFile, "not loaded successfully on all cores"), logFolderPath = logFolder)
+  }
+}
+
+#' @title loadSimulationOnCores
+#' @description Send libraryName to core, load the library and check that it has been loaded successfully
+#' @param libraryName string containing name of library to be loaded
+loadLibraryOnCores <- function(libraryName, logFolder = getwd()) {
+  success <- FALSE
+  Rmpi::mpi.bcast.Robj2slave(obj = libraryName)
+  Rmpi::mpi.remote.exec(library(libraryName, character.only = TRUE))
+  libraryLoaded <- Rmpi::mpi.remote.exec(checkLibraryLoaded(libraryName))
+  success <- checkAllCoresSuccessful(libraryLoaded)
+  validateIsLogical(success)
+  if (success) {
+    logWorkflow(message = paste(libraryName, "loaded successfully on all cores"), pathFolder = logFolder)
+  } else {
+    logErrorThenStop(message = paste(libraryName, "not loaded successfully on all cores"), logFolderPath = logFolder)
+  }
+}
+
+#' @title loadSimulationOnCores
+#' @description Send libraryName to core, load the library and check that it has been loaded successfully
+#' @param libraryName string containing name of library to be loaded
+updateIndividualParametersOnCores <- function(individualParameters, logFolder = getwd()) {
+  success <- FALSE
```

Comment:
No need to initialize to FALSE since it's reset right after

Reply:
Removed